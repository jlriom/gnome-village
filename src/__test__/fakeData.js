export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
