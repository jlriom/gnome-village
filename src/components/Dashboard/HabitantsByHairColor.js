import React from "react";

import CategoryItem from "./CategoryItem";

const HabitantsByHairColor = ({ habitantsByHairColor }) => (
	<ul className='list-group my-4'>
		<li className='list-group-item  list-group-item-secondary'>
			Habitants by hair color
		</li>
		{habitantsByHairColor.map(hairColor => (
			<CategoryItem
				category={hairColor.hair_color}
				count={hairColor.count}></CategoryItem>
		))}
	</ul>
);

export default HabitantsByHairColor;
