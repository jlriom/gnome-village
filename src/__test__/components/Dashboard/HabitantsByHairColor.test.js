import React from "react";
import HabitantsByHairColor from "../../../components/Dashboard/HabitantsByHairColor";
import renderer from "react-test-renderer";
import { habitantsByHairColor } from "../../fakeData";

describe("<HabitantsByHairColor />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <HabitantsByHairColor habitantsByHairColor={habitantsByHairColor} />
        );
        expect(tree).toMatchSnapshot();
    });
});

