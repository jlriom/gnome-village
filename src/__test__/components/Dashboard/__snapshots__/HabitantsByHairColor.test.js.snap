// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantsByHairColor /> renders ok 1`] = `
<ul
  className="list-group my-4"
>
  <li
    className="list-group-item  list-group-item-secondary"
  >
    Habitants by hair color
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Black
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        256
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Gray
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        258
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Green
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        267
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Pink
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        285
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Red
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        273
      </span>
    </span>
  </li>
</ul>
`;
